File: NHSOnline.Backend.Worker/Bridges/Emis/Models/Prescriptions/MedicationCourse.cs
using System.Collections.Generic;

namespace NHSOnline.Backend.Worker.Bridges.Emis.Models.Prescriptions
{
    public class MedicationCourse
    {
        public string MedicationCourseGuid { get; set; }

        public string Name { get; set; }

        public string Dosage { get; set; }

        public string QuantityRepresentation { get; set; }

        public PrescriptionType PrescriptionType { get; set; }

        public IEnumerable<string> Constituents { get; set; }

        public bool CanBeRequested { get; set; }
    }
}

File: bddtests/src/main/kotlin/models/prescriptions/MedicationCourse.kt
package models.prescriptions

import mocking.emis.models.PrescriptionType

data class MedicationCourse(
        val medicationCourseGuid : String,
        val name: String,
        val dosage: String?,
        val quantityRepresentation: String? = null,
        var prescriptionType: PrescriptionType? = null,
        val constituents: List<String>? = null,
        var canBeRequested: Boolean? = null)

{
    fun getInstructions() : ArrayList<String> {
        val instrictionsList = pages.prescription.resolveDetailsField(dosage, quantityRepresentation)
        return when (instrictionsList.size != 0){
            true -> instrictionsList
            false -> arrayListOf("")
        }
    }

    fun getInstructionsText() : String {
        return getInstructions().joinToString(" - ")
    }
}
